### ISSUE TITLE
A new way of ingesting nansat objects

### ISSUE THREAD

**Issue by mortenwh** [nansencenter/django-geo-spaas#32]
My problem is related to scatterometer data but the solution could be relevant in other cases as well. In geo-spaas, a full scatterometer acquisition will always(?) overlap with other data since the dataset covers a full orbit. The solution in the nansat mapper (`scatterometers.py`) is to split the scene into four. An extra kwarg `quartile` is therefore added, and its default value is 0, i.e., the first quartile of the scene is opened.

If we want to add all quartiles to geospaas, we need to create a separate management command but we also need to ensure that this is run every time a scatterometer dataset is ingested. Otherwise we will only ingest the first quartile.

Solution: Ensure the dataset is correctly ingested by adding a new method, named after the nansat mapper, before regular nansat ingestion. The nansat ingestor should always try to run such methods before falling back to default.

**Comment by akorosov** [comment]
You can also use option of the ingest command : --nansat-option quartile=1
Then it will call Nansat(filename, quartile=1)

On Aug 3, 2018 20:03, "Morten W. Hansen" <notifications@github.com> wrote:

My problem is related to scatterometer data but the solution could be
relevant in other cases as well. In geo-spaas, a full scatterometer
acquisition will always(?) overlap with other data since the dataset covers
a full orbit. The solution in the nansat mapper (scatterometers.py) is to
split the scene into four. An extra kwarg quartile is therefore added, and
its default value is 0, i.e., the first quartile of the scene is opened.

If we want to add all quartiles to geospaas, we need to create a separate
management command but we also need to ensure that this is run every time a
scatterometer dataset is ingested. Otherwise we will only ingest the first
quartile.

Solution: Ensure the dataset is correctly ingested by adding a new method,
named after the nansat mapper, before regular nansat ingestion. The nansat
ingestor should always try to run such methods before falling back to
default.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/nansencenter/django-geo-spaas/issues/32>, or mute the
thread

**Comment by mortenwh** [comment]
Yes, but then you'll have to know what you're doing.. If a user isn't aware
of this detail he or she will end up with only a fourth of the data..

Den lør. 4. aug. 2018, 10:41 skrev Anton Korosov <notifications@github.com>:

> You can also use option of the ingest command : --nansat-option quartile=1
> Then it will call Nansat(filename, quartile=1)
>

**Comment by akorosov** [comment]
I see. Isn't it simpler and more explicit to make a new ingestion command
for scaterrometers which runs (or inherits from) ingest four times for each
quartile?

**Comment by mortenwh** [comment]
Maybe - but can we ensure that this is run? The problem is that the nansat ingestor will anyway work on scatterometer data and ingest 1/4 of the data if a user is not aware of this specific issue...

**Comment by mortenwh** [comment]
I have looked at the code, and it is too complicated to generalize the `nansat_ingestor`. I anyway made some minor changes that need to be reviewed, and then I made a new package https://github.com/nansencenter/django-geo-spaas-scatterometry
